python/singa/layer.py

```diff
@@ -78,12 +78,54 @@ class Layer(object):
         name (str): layer name
     '''
 
-    def __init__(self, name, **kwargs):
-        self.layer = None  # layer converted by swig
-        self.name = name  # TODO(wangwei) duplicate with self.conf.name
-        self.conf = model_pb2.LayerConf()
-        self.conf.name = name
-        self.param_specs = []
+    def __init__(self, name, conf = None, **kwargs):
+        if conf == None:
+            self.layer = None  # layer converted by swig
+            self.name = name  # TODO(wangwei) duplicate with self.conf.name
+            self.conf = model_pb2.LayerConf()
+            self.conf.name = name
+            self.param_specs = []
+        else:
+            self.layer = None
+            self.name = conf.name
+            self.conf = conf
+            # TODO(Xiangrui) need to fill in param_specs
+            self.param_specs = []
+            if conf.type == 'Convolution' or conf.type == 4:
+                _check_engine(engine, ['cudnn', 'singacpp'])
```

Comment:
it seems all the if-else are for engine check?
we may add a function covert_engine(caffe_engine) -> signa_engine, and create each type of layer as 
```
eng = convert_engine(caffe_engine)
layer_type = convert_layer_layer(conf.type)
self.layer = __create_layer(engine, layer_type)
```
Dense layer needs special processing.